release-process.md

```diff
@@ -1,79 +1,97 @@
-
-## Remix libs release
-(remove dist and login to npm)
- - git fetch origin master
- - git checkout origin/master
- - git checkout -b bumpLibsVersion
- - yarn run publish:libs (this command uses lerna)
- - commit
-
-## Remix IDE release Part 1. First push master to beta. Feature Freeze
- - git co -b remix_beta origin/remix_beta
- - git reset --hard -master-commit-hash-
- - git push -f origin remix_beta
- 
-## Testing phase
-## In case of fixing bugs push PR's also to beta to include in Release
- 
-## Remix IDE release Part 2. Bump the version in beta and release
+This document includes the instructions to release for:
+ - remixd to NPM
+ - Remix libraries to NPM
+ - remix.ethereum.org
+ - remix-alpha.ethereum.org
+ - remix-beta.ethereum.org
 
- - git fetch origin remix_beta
- - git checkout origin/remix_beta
- - git checkout -b bumpVersion
- - update package.json version
- - update version in yarn.lock
- - merge PR to **origin/remix_beta**
- - git fetch origin remix_beta
- - git checkout origin/remix_beta
- - git tag v(version-number)
- - git push --tags
- - github-changes -o ethereum -r remix-project -a --only-pulls --use-commit-body --branch remix_beta --only-merges --between-tags previous_version...next_version
- - publish a release in github using the changelog
+## remixd NPM release
+ - Bump version for remixd in `./libs/remixd/package.json`
+ - Run: `nx build remixd`
+ - Move to build directory: `cd ./dist/libs/remixd`
+ - Publish to NPM: `npm publish` (Make sure you are logged in to NPM)
+ - create bump PR to master
+
+## Remix libraries NPM release
+ - Make sure you are on latest `master` branch
+ - `git pull origin master`
+ - `git checkout -b bumpLibsVersion`
+ - `yarn run publish:libs `
  
-## Remix IDE release Part 3. Bump dev branch (master)
+ This command uses `lerna` and solely responsible for publishing all the remix libraries. It will ask for a new version of each library. Make sure you are logged in to NPM.
 
- - git fetch origin master
- - git checkout origin/master
- - git checkout -b bumpDevVersion
- - update package.json version: bump the version and add the tag `dev` if not already present.
- - update version in yarn.lock
- - create a PR and merge it to origin/master
+ Once this command is done, versions for each remix library should be updated to latest in libs package.json file.
+ - Create bump PR to master
+
+## Feature Freeze
+Once feature freeze is done, `remix_beta` should be updated latest to the master which will automatically update `remix-beta.ethereum.org` through a CI job.
+
+ - `git checkout remix_beta`
+ - `git pull origin remix_beta`
+ - `git reset --hard <master-commit-hash>` (`master-commit-hash` will be latest commid id from `master` branch)
+ - `git push -f origin remix_beta`
  
-## Remix IDE release Part 4. remix.ethereum.org update
+## Testing Phase
+Testing is performed after the Feature Freeze. `build-qa-doc.js` script can be used to generate the list of QA tasks as described in the script itself.
```

Comment:
It might be good to elaborate a little more about how to run this script and then how this is used to update the google doc.

Reply:
instructions to use the file are given in the file itself : https://github.com/ethereum/remix-project/blob/master/build-qa-doc.js#L18

Reply:
added more details